### PR TITLE
Moves CLA bot config into this repo

### LINKED ISSUE #564: Store the `.clabot` here in the conda/infra repo
- [x] Move changes from clabot-config repo into infra repo, retaining the git history (https://github.com/conda/infra/pull/565)
- [x] Update the GHA workflow to use new location, do release (https://github.com/conda/actions/pull/42)
- [ ] Merge all PRs against the various repos to have new GHA workflow in use (TBD)
- [ ] Update README with note in clabot-config (https://github.com/conda/clabot-config/pull/5)
- [ ] Archive the clabot-config repo in https://github.com/conda-archive (TBD)

### PR DESCRIPTION
**Pull request by jezdez**
Fix #564.

This is based on commit https://github.com/conda/clabot-config/commit/c8ec6732d8ad6ddc8e9bb49faaa52bc0897b759e of the clabot-config repo.

The steps I took to do this:

```shell
git clone https://github.com/conda/clabot-config
git clon git@github.com:conda/infra.git conda-infra
cd conda-infra
git remote add src master ../clabot-config
git checkout -b clabot-config origin/clabot-config
git pull src master --allow-unrelated-histories
<fix merge conflicts in README and LICENSE>
git add -A
git commit
```

The change of the GitHub Actions workflow is in https://github.com/conda/actions/pull/42, which would require a new tagged release.

### CODE REVIEW

**Review by beeankha — APPROVED** [review]

File: check_and_sort.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+with open('.clabot') as f:
+    data = json.load(f)
+
+data['contributors'] = list(set(data['contributors']))
+data['contributors'].sort(key=lambda x: x.lower())
+
+with open('.clabot', 'w') as f:
+    json.dump(data, f, indent=2)
+    f.write("\n")
```
Comment (kenodegard) [inline check_and_sort.py:11]: Can we make this a pre-commit hook?

File: .clabot [hunk 2]
```diff
@@ -0,0 +1,521 @@
+{
+  "message": "We require contributors to sign our Contributor License Agreement, and we don't have one on file for {{usersWithoutCLA}}. In order for us to review and merge your code, please e-sign the PDF at https://conda.io/en/latest/contributing.html#conda-contributor-license-agreement.  We then need to manually verify your signature.  We will ping the bot to refresh the PR status when we have confirmed your signature.",
```
Comment (kenodegard) [inline .clabot:2]: This isn't used anywhere, can we remove it?

**Review by kenodegard — CHANGES_REQUESTED** [review]

File: check_and_sort.py (hunk 1 above)
Comment (jezdez) [reply check_and_sort.py:11]: Oooh, good idea!

File: .clabot (hunk 2 above)
Comment (jezdez) [reply .clabot:2]: I think so, it's a bit weird to support something obviously focused on the old clabot project anyways. Good call.

File: check_and_sort.py (hunk 1 above)
Comment (jezdez) [reply check_and_sort.py]: Done in https://github.com/conda/infra/pull/565/commits/0112893dafdd7b02d00298836de8aa15bdae49e6.

File: sort_json.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    def pairs_first(pairs: Sequence[tuple[str, str]]) -> Mapping[str, str]:
+        # Start of the modification
+        new = []
+        for name, value in pairs:
+            if name in sorted_keys:
+                value = list(set(value))
+                value.sort(key=lambda item: item.lower())
+            new.append((name, value))
+        pairs = new
+        # End of the modification
```
Comment (kenodegard) [inline sort_json.py:53]: Is this change perhaps something we can push upstream, if not it feels like a large script to keep around if all we want is this:

```python
from argparse import ArgumentParser
import json
from pathlib import Path
from typing import Optional, Sequence

def main(argv: Optional[Sequence[str]] = None) -> int:
    parser = ArgumentParser()
    parser.add_argument("--sort-keys", type=lambda v: v.split(","), default=[], help="Keys to sort")
    parser.add_argument("filenames", nargs="*", type=Path, help="Filenames to fix")
    args = parser.parse_args(argv)

    def object_pairs_hook(pairs):
        return {
            name: sorted(set(value), key=str.lower) if name in args.sort_keys else value
            for name, value in pairs
        }

    status = 0
    for path in args.filenames:
        original = path.read_text()
        try:
            content = json.loads(original, object_pairs_hook=object_pairs_hook)
        except ValueError:
            status = 2  # invalid json format
        else:
            pretty = f"{json.dumps(content, indent=2)}\n"  # include trailing newline
            path.write_text(pretty)
            status |= original != pretty  # record whether modifications were made
    return status

if __name__ == '__main__':
    raise SystemExit(main())
```

Comment (jezdez) [reply sort_json.py:53]: I'm fine just using what you posted, not thinking my updates are useful for upstream but I could be wrong.

Comment (jezdez) [reply sort_json.py]: Fixed in https://github.com/conda/infra/pull/565/commits/4e13e4d1d99eadbb1c1fd4c865a1fb6636f6dfbd.

**Comment by jezdez** [comment]
This should NOT be squashed when merging!

**Review by kenodegard — APPROVED** [review]

**Pull request merged**